xds/internal/balancer/cdsbalancer/cluster_handler.go

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cdsbalancer
+
+import (
+	"errors"
+	"sync"
+
+	xdsclient "google.golang.org/grpc/xds/internal/client"
+)
+
+var errNotReceivedUpdate = errors.New("tried to construct a cluster update on a cluster that has not received an update")
+
+// clusterHandler will be given a name representing a cluster. It will then update the CDS policy
+// constantly with a list of Clusters to pass down to XdsClusterResolverLoadBalancingPolicyConfig
+// in a stream like fashion.
+type clusterHandler struct {
+	// A mutex to protect entire tree of clusters.
+	clusterMutex    sync.Mutex
+	root            *clusterNode
+	rootClusterName string
+
+	// A way to ping CDS Balancer about any updates or errors to a Node in the tree.
```

Comment:
Now you have the plugin, wrap the comments. :)

Reply:
Almosstttttt doneeeeeeee...https://www.google.com/search?q=snorlax&source=lnms&tbm=isch&sa=X&ved=2ahUKEwiE2qClibjwAhVKneAKHcmlBo8Q_AUoAXoECAEQAw&biw=628&bih=1001#imgrc=S0FJs_X8a2idZM

Reply:
ITS DONE HALLELUJAH